### PR TITLE
[#1344] 1.7.3-hotfix-feature-res-warning-popup

### PR DESCRIPTION
**Pull request by zhiyuli**
This pr is a hotfix for issue #1344.

### CODE REVIEW

**Comment by zhiyuli** [comment]
@pkdash could you review this hotfix? thanks

**Comment by mjstealey** [comment]
> @pkdash could you review this hotfix

The error can be ignored as it's the [test_sax_parsing](http://ci.hydroshare.org:8080/job/github-pr-builder-hydroshare/265/testReport/junit/hs_core.tests.serialization.test_resourcemeta_sax_parsing/TestNetcdfResourceMetaSax/test_sax_parsing/)

File: hs_geographic_feature_resource/page_processors.py [hunk 1]
```diff
@@ -103,6 +103,7 @@ def landing_page(request, page):
         context['ori_coverage_form'] = ori_coverage_form
         context['geom_information_form'] = geom_information_form
 
+    context['edit_mode'] = edit_resource
```
Comment (pkdash) [inline hs_geographic_feature_resource/page_processors.py]: Not sure how this change fixes the problem.

Comment (zhiyuli) [inline hs_geographic_feature_resource/page_processors.py]: @pkdash the frontend is expecting to read this value to determine current mode: View or Edit.
If it is Edit mode, the frontend will show a custom warning popup if users try to delete a mandatory file from the Feature Res

Comment (pkdash) [inline hs_geographic_feature_resource/page_processors.py]: Can the user delete a file in the view mode?

Comment (zhiyuli) [inline hs_geographic_feature_resource/page_processors.py]: @pkdash no, user cannot delete a file in View mode. The deletion icon only shows up in Edit mode, and when user clicks that deletion icon a default warning popup will show up for any res type, in feature res type we hacked this default popup, replacing it with our customized popup that tells user "delete any of the mandatory shapefile will remove all shapefiles".  To do this, we need to let the frontend know what mode it is in: View or Edit.

Comment (pkdash) [inline hs_geographic_feature_resource/page_processors.py]: I am not still sure why the front end needs to know view/edit mode. Why can't you display the popup when the user tries to delete the file. May be I am missing something. After all it is a small change. So +1.

**Pull request merged**